Pixel Cluster Counting Automation

samhiggie opened the pull request:
Working Update to PCC Producer workflow in 9_1_X, trimmed files and added ZeroBias and Random Trigger configuration files to test run. 
The test configuration files are located here:
CMSSW_9_1_0_pre1/src/Calibration/PCCAlCaRecoProducers/test/PCC_Random_cfg.py
CMSSW_9_1_0_pre1/src/Calibration/PCCAlCaRecoProducers/test/PCC_ZeroBias_cfg.py 

The Lumisection sample rate can be changed in each of these configuration files, but the defaults are 100LS for Random and 1 for ZeroBias. 
_____________________________________________________________________________________________________________
Major Steps to Project Completion:
1. Construction of rawPCCProducer that will loop over the modules and count the pixel clusters 
2. Harvesters to save reco::PCC object (output from AlcaPCCProducer) and LumiInfo object (output from rawPCCProducer)

Other collaborators working directly on this project:
@capalmer85

samhiggie committed:
Working Update to AlcaPCCProducer workflow in 9_1_X, trimmed files and added ZeroBias and Random Triggers

samhiggie committed:
Some comments to test GH

samhiggie committed:
AlCaPCCProducer Workflow in 9_1_X with ZB and Random Trig.

samhiggie committed:
AlcaPCCProducer 9_1_X ZB and Rand Fixed Classes.h

cmsbuild commented:
A new Pull Request was created by  @samhiggie  for master.

It involves the following packages:

Calibration/PCCAlCaRecoProducers
DataFormats/Luminosity


The following packages do not have a category, yet:

Calibration/PCCAlCaRecoProducers
Please create a PR for https://github.com/cms-sw/cms-bot/blob/master/categories.py to assign category

@perrotta, @cmsbuild, @slava77, @davidlange6 can you please review it and eventually sign? Thanks.
@muzaffar, @davidlange6, @smuzaffar you are the release manager for this.

cms-bot commands are listed here cms-sw/cmssw#13028

slava77 commented on DataFormats/Luminosity/src/classes_def.xml:
```diff
@@ -40,4 +40,11 @@
+   <version ClassVersion="13" checksum="1593740538"/>
+   <version ClassVersion="12" checksum="3216246807"/>
+   <version ClassVersion="11" checksum="1497064974"/>
+   <version ClassVersion="10" checksum="915260007"/>
```
please start from 3 and drop all intermediate development versions

slava77 commented on DataFormats/Luminosity/interface/PCC.h:
```diff
@@ -0,0 +1,74 @@
+        }
+
+      private:
+        std::vector<std::vector<int> > m_counts;
```
vectors of vectors should be avoided in data formats.
What are the typical dimensions?
It may be better to flatten this out

capalmer85 replied:
We need pixel cluster counts with module ID x BCID granularity.  We like this scheme because we do not need to know the module IDs beforehand.  I think the only way to flatten would be to hardcode all the module IDs beforehand (which isn't a nice option).

Do you have an alternative suggestion?

slava77 replied:
you can change m_counts to 1-D vector and add a vector with indices inside m_counts.
That's all in case BCID range varies, if it's a fixed length, then you only need an extra into to count the number of module IDs.

slava77 commented on DataFormats/Luminosity/interface/PCC.h:
```diff
@@ -0,0 +1,74 @@
+            m_events[BXid]++;
+        }
+        ////////////////////////////////////////////
+        std::vector<std::vector<int>> const & read_counts() const {
```
please use consistent naming of methods: readCounts matches the rest

slava77 commented on DataFormats/Luminosity/interface/PCC.h:
```diff
@@ -0,0 +1,74 @@
+    public:
+        PCC() : m_events(nEmBX){}
+        ////////////////////////////////////////////
+        void Increment(int mD,int BXid,int count){
```
start local variables and data members with lower case letter. Capitalization is reserved for class names.

slava77 commented on DataFormats/Luminosity/interface/PCC.h:
```diff
@@ -0,0 +1,74 @@
+#ifndef PCC_h
+#define PCC_h
```
Please use an include guard matching the full file name subsystem_package_fileName_h

slava77 commented on DataFormats/Luminosity/interface/PCC.h:
```diff
@@ -0,0 +1,74 @@
+#include <vector>
+
+namespace reco {
+class PCC {
```
A less abbreviated name would be better to avoid possible symbol clashes

samhiggie committed:
Updates to PCC.h members and classes_def.xml

samhiggie commented:
I've changed m_counts to a single vector<int>, renamed a function, and cleaned up the classes_def.xml. Will implement the other recommendations soon.

samhiggie committed:
Requested Changes to PCC workflow implemented

cmsbuild commented:
Pull request #18233 was updated. @perrotta, @cmsbuild, @slava77, @davidlange6 can you please check and sign again.

samhiggie commented on Calibration/PCCAlCaRecoProducers/plugins/AlcaPCCProducer.cc:
```diff
@@ -0,0 +1,115 @@
+//--------------------------------------------------------------------------------------------------
+void AlcaPCCProducer::endLuminosityBlock(edm::LuminosityBlock const& lumiSeg, const edm::EventSetup& iSetup){
+    //Saving the PCC object 
+    //FIXME! The line below doesn't work but I need to save here. 
```
I would like to save the object here because as of now the increment method is lost when resetNLumi_ isn't reached at the end of the set of lumi sections.  Example: I have 20 LS and I define my block at 7. So far I save 1-14 and 15-20 isn't saved.

slava77 replied:
if you want to write to the LumiBlock, use endLuminosityBlockProduce.
I see it's used below already.

capalmer85 replied:
The problem is we need to be able to save every several lumi sections for some of the data.  We have this line to allow the .put():

 if (countLumi_%resetNLumi_!=0) return;

In the case that Sam is talking about, we have a file with 20 LSs and we want to save every 7 LSs.  (Arbitrary number to highlight the issue.)  It saves the counts from LSs 1-7 after LS 7 and from LSs 8-14 after LS 14, but LSs 15-20 are never saved.

We were thinking that we need an additional .put() line some place between the processing of the final LS and the termination of the program.  (Or is there some way to know that you are at the final LS?)

slava77 replied:
I think that you still need to write an object every lumi, possibly an empty one.

@Dr15Jones, please advise if you have a more fitting case in mind

Dr15Jones replied:
It would be better to write an object every lumi.

capalmer85 replied:
The beam spot producer writes out every N LSs with similar code... I don't see why we can't.
https://github.com/cms-sw/cmssw/blob/master/Calibration/TkAlCaRecoProducers/plugins/AlcaBeamSpotProducer.cc#L100

There is some context for our producer below.  Can you help us with the question as posed?  If you aren't sure, then we can ping the authors of the beam spot producer since they must have the same issue.


We will have two parallel workflows:
1) zero bias - this will be written every LS
2) random triggers - we are not able to take enough data to make a meaningful measurement per LS.  We typically make a measurement grouping 100 LSs.  

This is the current offline workflow that we are replacing and for 2) we will have to merge these objects anyway.  Merging later in the production chain will give us the same issue but in a different producer, so I don't see an advantage to waiting.  (If you do, please motivate us.)

capalmer85 replied:
@slava77 @Dr15Jones We would really like to keep this workflow and be able to sum over LSs here.  Could you suggest an appropriate solution please?

slava77 replied:
You can write something "empty" or a running average since the last "complete and useful" entry.
The strategy to occasionally save something in a lumi seems to me to fit better in the run product.

Dr15Jones replied:
My personal opinion is you'd probably better be served by writing a summary out each lumi and then have another job which runs over the file to do your final agregation. That would also prepare you for the time when the framework will process multiple luminosity blocks simultaneously (which hopefully will happen by the end of Summer).

As for the beam spot, I checked and all non-test uses of the beam spot producer I could find write each luminosity block. See
https://github.com/cms-sw/cmssw/search?utf8=✓&q=fitEveryNLumi&type=

capalmer85 replied:
@slava77 Using run product is an interesting idea.  Does that ensure that LSs are visited in order?  Is that true for the lumi-aware event processing (ie as it is now)?

@Dr15Jones Is this the same module that is used for delivering the beam spot in prompt reco?  I know from experience that it is not delivered every lumi section.  (I can forward you references.)

samhiggie replied:
After this discussion, Chris and I have decided to write to the object every lumi. The overall structure of the producer should not change much. Please stay tuned for further updates.

samhiggie commented:
All the comments above have been implemented. Question moving forward: How can I save my object at the end of the luminosity block? Please see detailed comment in the AlcaPCCProducer.cc file line 96.

slava77 commented on Calibration/PCCAlCaRecoProducers/interface/AlcaPCCProducer.h:
```diff
@@ -0,0 +1,55 @@
+
+#include "DataFormats/Luminosity/interface/PCC.h"
+
+class AlcaPCCProducer : public edm::one::EDProducer<edm::EndLuminosityBlockProducer,
```
Can this be made ```edm::global``` or ```edm::stream```?
See examples in 
https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkStreamModuleInterface#edm_EndLuminosityBlockProducer
or https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkGlobalModuleInterface#edm_EndLuminosityBlockProducer

capalmer85 replied:
Probably... from the twiki once up from these I see:

A Stream module is one which only cares about Events individually and not as a group and therefore works just fine if it only sees a subset of the Events coming from the Source.

and 

A Global module is one which must see all Events coming from a Source in order to do its job. 

In our case each event is processed individually but only saves at the end of a lumi section (or series of lumi sections). So conceptually I don't see where this producer fits most appropriately.  Maybe you could educate us about which type would be best for this producer and highlight the benefits for it?

slava77 replied:
both twikis have an example.
I think that edm::stream will match the current implementation a bit more easily.
We should avoid edm::one if possible

Dr15Jones replied:
Using a seperate .h file is no longer recommended for modules. Please move the code into the .cc file.

slava77 commented on DataFormats/Luminosity/interface/PCC.h:
```diff
@@ -0,0 +1,56 @@
+#ifndef PCC_PCC_h
+#define PCC_PCC_h
```
Please use an include guard matching the full file name subsystem_package_fileName_h

slava77 replied:
Also, please rename the file to match the primary class name defined in the file

slava77 commented on DataFormats/Luminosity/src/classes_def.xml:
```diff
@@ -40,4 +40,8 @@
   <class name="std::vector<LumiSummary::L1>"/>
   <class name="std::vector<LumiSummary::HLT>"/>
+  <class name="reco::PixelClusterCounts" ClassVersion="15">
+   <version ClassVersion="15" checksum="1280308955"/>
```
please start from version 3

capalmer85 replied:
@samhiggie change '15' to '3'

slava77 commented:
assign alca

cmsbuild commented:
New categories assigned: alca

@mmusich,@cerminar,@arunhep,@franzoni,@ghellwig you have been requested to review this Pull request/Issue and eventually sign? Thanks

Dr15Jones commented on DataFormats/Luminosity/src/classes.h:
```diff
@@ -14,5 +15,7 @@ namespace DataFormats_Luminosity {
       edm::Wrapper<LumiInfoRunHeader> lumiinforunheaderobj;
       edm::Wrapper<LumiInfo> lumiinfoobj;
       edm::Wrapper<BeamCurrentInfo> beamcurrentinfoobj;
+      reco::PixelClusterCounts b;
```
This line is not needed.

slava77 commented:
@samhiggie 
please follow up on the code review comments
Thank you.

samhiggie committed:
Minor fixes and saves every lumi

cmsbuild commented:
Pull request #18233 was updated. @perrotta, @cmsbuild, @slava77, @davidlange6 can you please check and sign again.

samhiggie committed:
Further cleaning and removing unnecessary files

cmsbuild commented:
Pull request #18233 was updated. @perrotta, @cmsbuild, @slava77, @davidlange6 can you please check and sign again.

samhiggie committed:
Further cleaning and removing unnecessary files

cmsbuild commented:
Pull request #18233 was updated. @perrotta, @cmsbuild, @slava77, @davidlange6 can you please check and sign again.

slava77 commented:
@samhiggie 
please post a message when you are done with commits

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

slava77 commented:
@samhiggie 
please clarify if you are done with updates of this PR so far.
Thank you.

cmsbuild commented:
+1
Tested at: 5a3db82ccb78871eab42020b6e3dd8fd5a9a4b6e
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18233/19342/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18233/19342/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 1780042
 - DQMHistoTests: Total failures: 29875
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1749994
 - DQMHistoTests: Total skipped: 173
 - DQMHistoTests: Total Missing objects: 0
 - Checked 94 log files, 14 edm output root files, 23 DQM output files

samhiggie commented:
Hello, I am still trying to integrate the comment on changing from edm::one to edm::stream. So the PR is not fully complete. However, even with the example listed I am still having trouble converting it to the new namespace.

Dr15Jones commented:
Could you give some specifics as to the problems you are having?

samhiggie commented:
The precise error is: error: cannot call member function 'virtual void AlcaPCCProducer::globalEndLuminosityBlockProduce(edm::LuminosityBlock&, const edm::EventSetup&, const LuminosityBlockContext*)' without object
         T::globalEndLuminosityBlockProduce(Lumi,iES,iRC);
I had to change the from the method "endLuminosityBlockProduce" to "globalEndLuminosityBlockProduce" as indicated in the wiki:
https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkStreamModuleInterface#edm_EndLuminosityBlockProducer

Dr15Jones commented:
I believe the problem is you must declare `globalEndLuminosityBlockProduce` to be a `static` class function.

samhiggie commented:
The only standing issue right now is changing to the edm::stream namespace. I can see from the examples that it might be to our benefit to structure everything out of the class definition, but I have yet to do so. Would it be alright to merge this PR as is and submit a new one with this hyperthreading feature in the near future? I will be traveling tomorrow, but I am confident I can incorporate these changes by the end of the week.

slava77 commented:
@samhiggie 
did you try the last suggestion in https://github.com/cms-sw/cmssw/pull/18233#issuecomment-296804769
?

samhiggie commented:
Yes I did try the suggestion of denoting the function as static in it's definition, but it did not work. I kept the declaration and the definition of the functions separate which I believe might be the trouble. In all the examples they place the declaration and definition into the class. I am currently working on implementing the latter.

samhiggie commented:
Due to nature of the rewrite for this hyper-threading feature, I have opened a new branch and made a commit: 
https://github.com/cms-sw/cmssw/commit/6386a5adcf8ebe4bec0c4dcb2ad4ff6c45ae1ae4
The code doesn't compile, but I think a more extensive rewrite may be in order so I would like to discuss that. 
Would it be a good idea to merge this PR and start developing this new feature elsewhere?

capalmer85 commented:
We need a version of this committed for commissioning in May. I am on paternity leave right now but will return soon. Please merge this branch without the multithreading and I will personally make the update in May before we start taking substantial data.

@slava77 @dr15jones is that ok?

capalmer85 commented:
@slava77 @Dr15Jones There are other aspects of this workflow that @samhiggie needs to work on right away, and this PR is essential for the new workflow which we need in place ASAP.

Could you let us know when the next CMSSW release milestone is?  In any event, I think it would be best for Sam to move on to the next phases and I do the one-to-stream conversion when I'm back from leave.

davidlange6 commented on DataFormats/Luminosity/interface/PixelClusterCounts.h:
```diff
@@ -0,0 +1,56 @@
+            size_t modIndex = it - m_ModID.begin();
+
+            if(it == m_ModID.end()){
+                std::vector<int> m_empBX(nEmBX, 0);
```
Hi @samhiggie - i had a quick look at this PR - I doubt this line is doing what you want it to. What do you mean to accomplish here?

samhiggie replied:
Here I am initializing a vector that has a number of 0s equal to the number of empty bunch crossings. This is appended to the m_counts vector every time a new module is introduced, which is later filled with ints that count the number of clusters at the BXid.

slava77 commented on Calibration/PCCAlCaRecoProducers/plugins/AlcaPCCProducer.cc:
```diff
@@ -0,0 +1,133 @@
+        }
+        int nCluster = mod.size();
+        if(nCluster!=nClusterCount) {
+            std::cout<<"counting yields "<<nClusterCount<<" but the size is "<<nCluster<<"; they should match."<<std::endl;
```
this has to be converted to LogWarning or LogInfo or hidden behind "if (debug)" depending on frequency of occurrence (rough thresholds, per event for the first two are not more frequent than O(0.01), O(1))

slava77 commented on Calibration/PCCAlCaRecoProducers/plugins/AlcaPCCProducer.cc:
```diff
@@ -0,0 +1,133 @@
+    endLumiOfPCC_ = lumiSeg.luminosityBlock();
+
+    //Saving the PCC object 
+    //std::cout<<"Saving Object "<<std::endl;
```
better remove or compile it with a configurable "if (debug)" or similar

slava77 commented on Calibration/PCCAlCaRecoProducers/test/PCC_Random_cfg.py:
```diff
@@ -0,0 +1,210 @@
+        'keep *_MEtoEDMConvertSiStrip_*_*')
+)
+#Make sure that variables match in producer.cc and .h
+process.alcaPCCProducer = cms.EDProducer("AlcaPCCProducer",
```
is the AlcaPCCProducer expected to run in a centrally supported workflow?
I guess that in that case this PR needs an appropriate default config fragment in a /python subdir somewhere.

samhiggie replied:
Yes I think it is expected to run in a central workflow. Do I need to make files similar to 
https://github.com/cms-sw/cmssw/blob/master/Calibration/TkAlCaRecoProducers/python/AlcaBeamSpotProducer_cfi.py 
Along with _cff.py? If so, how can I specify the High Level trigger so I can select between Random and ZeroBias HLT filters?

slava77 commented on DataFormats/Luminosity/interface/PixelClusterCounts.h:
```diff
@@ -0,0 +1,56 @@
+
+namespace reco {
+class PixelClusterCounts {
+static constexpr unsigned int nEmBX=3564;//Empty BX to fill with counts
```
better spell it out "nEmptyBX"

slava77 replied:
this can apparently be taken directly from DataFormats/Luminosity/interface/LumiConstants.h
```static const unsigned int numBX = 3564;       // number of BX per orbit```

slava77 commented on DataFormats/Luminosity/interface/PixelClusterCounts.h:
```diff
@@ -0,0 +1,56 @@
+    public:
+        PixelClusterCounts() : m_events(nEmBX){}
+        ////////////////////////////////////////////
+        void increment(int mD,int BXid,int count){
```
here and below please add doxygen style comments to be used for code self-documentation ( comment text should be prefixed  with ```///``` or ```//!``` )

slava77 commented on DataFormats/Luminosity/interface/PixelClusterCounts.h:
```diff
@@ -0,0 +1,56 @@
+            m_counts[nEmBX*modIndex+BXid] += count; 
+        }
+        ////////////////////////////////////////////
+        void eventCounter(int BXid){
```
bxID would not break CMSSW naming style (first letter capitalization of for class types)

slava77 replied:
```incrementEventCounter``` seems more intuitive

slava77 commented on DataFormats/Luminosity/interface/PixelClusterCounts.h:
```diff
@@ -0,0 +1,56 @@
+        void increment(int mD,int BXid,int count){
+            std::vector<int>::iterator it;
+            it = std::find(m_ModID.begin(), m_ModID.end(), mD);
+            size_t modIndex = it - m_ModID.begin();
```
```c++
 size_t modIndex = std::distance(m_ModID.begin(), std::find(m_ModID.begin(), m_ModID.end(), mD));
 if (modIndex == m_ModID.size()){
     m_ModID.push_back(mD);
     m_counts.resize(m_counts.size()+nEmBX, 0);
}
```

Please add a constructor with a number of modules to be reserved (is it 2528 for the phase-1 total? ) so that we do not have massive reallocations going on.
I suppose that during a run this m_counts vector eventually grows to about 36MB in memory.
Is it correct?

samhiggie replied:
That estimated size is correct. However I don't think we can handle the module IDs at this point, which I think we would need to do if we implemented this feature. We designed this with the idea in mind that not all modules may not be used in a lumisection.

slava77 commented on DataFormats/Luminosity/interface/PixelClusterCounts.h:
```diff
@@ -0,0 +1,56 @@
+            if(it == m_ModID.end()){
+                std::vector<int> m_empBX(nEmBX, 0);
+                m_ModID.push_back(mD);
+                m_counts.insert(m_counts.begin(),m_empBX.begin(),m_empBX.end()); 
```
the count increment is done below for the module at index corresponding to the end of the vector module, why is this insert made in front?
It looks incorrect.
My suggestion above should probably replace this

slava77 commented on DataFormats/Luminosity/interface/PixelClusterCounts.h:
```diff
@@ -0,0 +1,56 @@
+        }
+        ////////////////////////////////////////////
+        void eventCounter(int BXid){
+            m_events[BXid]++;
```
there should probably be an assert that BXid is within range.
It can be achieved with ```m_events.at(BXid)++;```

samhiggie replied:
Working on this, there are some BXids that throw the error that it appears to be out of range. May implement a try-block to catch exceptions.

slava77 commented on DataFormats/Luminosity/interface/PixelClusterCounts.h:
```diff
@@ -0,0 +1,56 @@
+        std::vector<int> m_counts;
+        std::vector<int> m_events;
+        std::vector<int> m_ModID;
+        std::vector<int> m_empBX;
```
this looks superfluous

franzoni commented:
hello @capalmer85 @samhiggie 
it would help us AlCa/Db if you could answer the questions 
we discussed at during the talk, also listed here
https://indico.cern.ch/event/626461/attachments/1445028/2225883/AlCa-DBMintuesfor10thApr2017.pdf

Which workflow are you aiming at including this producer in ?

capalmer85 commented:
On Fri, Apr 28, 2017 at 8:23 AM, Giovanni Franzoni <notifications@github.com
> wrote:

> hello @capalmer85 <https://github.com/capalmer85> @samhiggie
> <https://github.com/samhiggie>
> it would help us AlCa/Db if you could answer the questions
> we discussed at during the talk, also listed here
> https://indico.cern.ch/event/626461/attachments/1445028/2225883/AlCa-
> DBMintuesfor10thApr2017.pdf
>
Will do next week.  Most of the questions about the veto list, which is not
an input for this step in the process.


> Which workflow are you aiming at including this producer in ?
>
This should run as a module in the ALCALUMIPIXELS stream.  Our intent is to
save these sums once in the promptest manner possible and eventually it
will replace the ALCALUMIPIXELS ALCARECO output.  The output from this
producer will be our raw data.

We will run the next producer, rawPCCProducer, once promptly and once more
at the end of the year with an updated veto module list.

> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/18233#issuecomment-297984419>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AFKqh1IlZ-AAIrx1swiX7nknrVQOQh9aks5r0dpcgaJpZM4M1O6C>
> .
>

samhiggie committed:
Clean Up files

cmsbuild commented:
Pull request #18233 was updated. @perrotta, @cmsbuild, @slava77, @davidlange6 can you please check and sign again.

samhiggie committed:
Added Config Fragment and fixed bxID counting, management etc.

cmsbuild commented:
Pull request #18233 was updated. @perrotta, @cmsbuild, @slava77, @davidlange6 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 9d39217d15b785db55a4b80d7638fc569d83e14d
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18233/19522/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18233/19522/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 3312 differences found in the comparisons
 - DQMHistoTests: Total files compared: 24
 - DQMHistoTests: Total histograms compared: 1831629
 - DQMHistoTests: Total failures: 32701
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1798748
 - DQMHistoTests: Total skipped: 180
 - DQMHistoTests: Total Missing objects: 0
 - Checked 98 log files, 14 edm output root files, 24 DQM output files

franzoni commented:
+1

capalmer85 committed:
Merged PCC_9_1_X from repository samhiggie

slava77 commented:
@franzoni 
please confirm that Calibration/PCCAlCaRecoProducers package name and structure is OK for AlCa.
Once confirmed we can edit the categories file https://github.com/cms-sw/cms-bot/blob/master/categories.py to have it appropriately assigned to AlCa next time it is updated.

davidlange6 commented:
LumiAlCaRecoProducers is probably a better name than PCCAlCaRecoProducers
> On May 8, 2017, at 3:18 PM, Slava Krutelyov <notifications@github.com> wrote:
> 
> @franzoni
> please confirm that Calibration/PCCAlCaRecoProducers package name and structure is OK for AlCa.
> Once confirmed we can edit the categories file https://github.com/cms-sw/cms-bot/blob/master/categories.py to have it appropriately assigned to AlCa next time it is updated.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

slava77 commented on Calibration/PCCAlCaRecoProducers/python/AlcaPCCProducer_cfi.py:
```diff
@@ -0,0 +1,9 @@
+import FWCore.ParameterSet.Config as cms 
+
+alcaPCCProducer = cms.EDProducer("AlcaPCCProducer",
```
cfi file name should better match with the module instance defined in it: alcaPCCProducer_cfi.py

slava77 commented on DataFormats/Luminosity/interface/PixelClusterCounts.h:
```diff
@@ -14,40 +14,39 @@
-            m_events[BXid]++;
+
+        void eventCounter(unsigned int  bxID){
+                m_events.at(bxID-1)++;
```
Just to be sure, the bxID counting starts from 1, right?

samhiggie replied:
The bxID does start at 1. Reran over file and printed out the bxID just to make sure.

slava77 commented:
current status : 
- (not reco)
    - need to move the producer code to Calibration/LumiAlCaRecoProducers
    - probably rename the cfi 
- for reco:
    - need a confirmation of the range of bxID values

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/18233	9d39217
- reco-related changes (lumi data format) look OK
- jenkins tests pass and comparisons show no differences

Quite likely this will need to be signed again after pending changes in non-reco code

capalmer85 committed:
add ALCARECO workflows for ZeroBias and Random triggers in ALCASTREAM and RECO

samhiggie committed:
Merge pull request #1 from capalmer85/merge-to-sam-may8

Merge to sam may8

cmsbuild commented:
Pull request #18233 was updated. @perrotta, @ghellwig, @arunhep, @cerminar, @cmsbuild, @franzoni, @slava77, @mmusich, @davidlange6 can you please check and sign again.

capalmer85 commented:
@slava77, @davidlange6 the two "not reco" comments are addressed here.

Anyone know who in operations will review and sign?

@franzoni As we discussed last Friday, I setup the new workflows for expected alcarecos.  I tested all of them with these cmsDriver cmds, which gave valid, filled output:

cmsDriver.py alcaPCCRandomFromRECO  --datatier ALCARECO --conditions auto:run1_data -s ALCA:AlCaPCCRandomFromRECO --eventcontent ALCARECO -n 100 --filein /store/data/Run2016B/ZeroBias/RECO/PromptReco-v2/000/273/158/00000/185656A3-F019-E611-9018-02163E0118D6.root
cmsDriver.py alcaPCCZeroBiasFromRECO  --datatier ALCARECO --conditions auto:run1_data -s ALCA:AlCaPCCZeroBiasFromRECO --eventcontent ALCARECO -n 100 --filein /store/data/Run2016B/ZeroBias/RECO/PromptReco-v2/000/273/158/00000/185656A3-F019-E611-9018-02163E0118D6.root
cmsDriver.py alcaPCCRandom  --datatier ALCARECO --conditions auto:run1_data -s ALCA:AlCaPCCRandom --eventcontent ALCARECO -n 100 --filein  /store/data/Run2016H/AlCaLumiPixels1/RAW/v1/000/281/640/00000/889664D8-C983-E611-9334-02163E0133B6.root
cmsDriver.py alcaPCCZeroBias  --datatier ALCARECO --conditions auto:run1_data -s ALCA:AlCaPCCZeroBias --eventcontent ALCARECO -n 100 --filein /store/data/Run2016H/AlCaLumiPixels1/RAW/v1/000/281/640/00000/889664D8-C983-E611-9334-02163E0133B6.root

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: ef607cef9e9fb1caf944ea3b753c8ff6935c3470
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18233/19652/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/ef607cef9e9fb1caf944ea3b753c8ff6935c3470
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18233/19652/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18233/19652/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18233/19652/summary.html

There are some workflows for which there are errors in the baseline:
 23234.0 step 3
 The results for the comparisons for these workflows could be incomplete 
This means most likely that the IB is having errors in the relvals.The error does NOT come from this pull request

Comparison Summary:
 - You potentially added 2460 lines to the logs
 - Reco comparison results: 3366 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 1777976
 - DQMHistoTests: Total failures: 32697
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1745106
 - DQMHistoTests: Total skipped: 173
 - DQMHistoTests: Total Missing objects: 0
 - Checked 94 log files, 14 edm output root files, 23 DQM output files

slava77 commented on Calibration/LumiAlCaRecoProducers/python/ALCARECOAlCaPCCZeroBiasFromRECO_Output_cff.py:
```diff
@@ -0,0 +1,16 @@
+        SelectEvents = cms.vstring('pathALCARECOAlCaPCCZeroBiasFromRECO')
+    ),
+    outputCommands = cms.untracked.vstring(
+        'keep *_*alcaPCCProducer*_*_*') 
```
do any of these get added to the standard workflows?
We should avoid as much as possible wildcards in the module instance names to avoid making unnecessary modules to be run in the unscheduled setup.
This seems to have specific enough name that could probably be left as is.

slava77 replied:
@Dr15Jones 
is it even happening with the task feature now enabled? Would a dangling loaded module be run triggered by a keep statement if it's not already explicitly listed on a task?

Dr15Jones replied:
Only modules on a Task will be run unscheduled.

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/18233	ef607ce
- updates since the last reco signoff are in non-reco areas https://github.com/cms-sw/cmssw/pull/18233/files/9d39217d15b785db55a4b80d7638fc569d83e14d..ef607cef9e9fb1caf944ea3b753c8ff6935c3470
    - code moved to Calibration/LumiAlCaRecoProducers
    - standard fragments are added in Configuration/StandardSequences/python/AlCaRecoStreams_cff.py to be available in cmsDriver ALCA:XYZ
- jenkins tests pass; the new code is not running by default yet

slava77 commented:
https://github.com/cms-sw/cms-bot/pull/842
to resolve the new-package-pending

franzoni commented on Calibration/LumiAlCaRecoProducers/python/ALCARECOAlCaPCCRandomFromRECO_cff.py:
```diff
@@ -0,0 +1,19 @@
+import HLTrigger.HLTfilters.hltHighLevel_cfi
+ALCARECORandomFromRECOHLT = HLTrigger.HLTfilters.hltHighLevel_cfi.hltHighLevel.clone(
+    HLTPaths = cms.vstring("*Random*"),
+    eventSetupPathsKey='',
```
hello @capalmer85 

@trtomei and @tocheng 
are in the process of configuring every alcareco to be steerable via triggerbits  for the HLT selection they do, which will ease changes in operations if you'll want to tune the input to your producer.
Please keep in mind we'll want to do that once this PR is merged
( let's not do it now not to couple two rather complex PR's)

franzoni commented:
@slava77 
good for AlCathe package naming you and @davidlange6 chose

@capalmer85 
thanks for the follow up to last Friday
it would marvellous if you could turn your 4 cmsDriver of here above
into four relval simple workflows to end up here https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_production.py (this is not a requirement - something we should do in this PR or asap otherwise)

ok both for alca and operations

franzoni commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

franzoni commented:
@capalmer85
I'd suggest you start preparing the back port of this to the 91x

davidlange6 commented on Calibration/LumiAlCaRecoProducers/plugins/AlcaPCCProducer.cc:
```diff
@@ -0,0 +1,121 @@
+        DetId detId = mod.id();
+
+        // -- clusters on this det
+        edmNew::DetSet<SiPixelCluster>::const_iterator  di;
```
Hi @samhiggie 

can this not be shortened to 
unsigned int nClusterCount = mod.size();
?

davidlange6 commented on Calibration/LumiAlCaRecoProducers/plugins/AlcaPCCProducer.cc:
```diff
@@ -0,0 +1,121 @@
+            nClusterCount++;
+        }
+        int nCluster = mod.size();
+        thePCCob->increment(detId(), bx, nCluster);
```
ok, seems the answer is yes... perhaps the stanza about nClusterCount is obsolete?

capalmer85 replied:
This was supposed to be part of a sanity check.  We can comment the loop if you like.  Do you want that done in this PR?  If we do that, everyone has to sign off again...

davidlange6 replied:
what sanity check? its computing the same information twice unless you are worried about the DetSet iterator not functioning like it should?

davidlange6 commented:
otherwise, please add this to one of the tests in runTheMatrix (perhaps 1001)?

capalmer85 commented:
@davidlange6:  @franzoni also mentioned adding the tests into relval_production.py but in a separate PR.  Can we also add the runTheMatrix test there too?

I'm just concerned that making minor changes will trigger another round of sign offs and potentially more delay... we are already late and need to back port.

franzoni commented:
I too would suggest to decouple the development of relval test.

franzoni commented:
transferring a small block to the T2_CH_CERN relval
for relval to be  developed
https://cmsweb.cern.ch/phedex/prod/Request::View?request=1003291

samhiggie committed:
Commented out cluster stanza

cmsbuild commented:
Pull request #18233 was updated. @perrotta, @ghellwig, @arunhep, @cerminar, @cmsbuild, @franzoni, @slava77, @mmusich, @davidlange6 can you please check and sign again.

slava77 commented on Calibration/LumiAlCaRecoProducers/plugins/AlcaPCCProducer.cc:
```diff
@@ -88,11 +88,11 @@ void AlcaPCCProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
+        //int nClusterCount=0;
+        //for (di = mod.begin(); di != mod.end(); ++di) {
+        //    nClusterCount++;
+        //}
```
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant

samhiggie replied:
We plan to use this later when we add a threshold for the pixels. Will place an explanation in comment and then push again.

samhiggie committed:
Commented out lines and added description

cmsbuild commented:
Pull request #18233 was updated. @perrotta, @ghellwig, @arunhep, @cerminar, @cmsbuild, @franzoni, @slava77, @mmusich, @davidlange6 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: e54455992c91638caef839e3ab2eee6787be9e37
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18233/19854/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18233/19854/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 3362 differences found in the comparisons
 - DQMHistoTests: Total files compared: 24
 - DQMHistoTests: Total histograms compared: 1829394
 - DQMHistoTests: Total failures: 62558
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1766656
 - DQMHistoTests: Total skipped: 180
 - DQMHistoTests: Total Missing objects: 0
 - Checked 98 log files, 14 edm output root files, 24 DQM output files

franzoni commented:
+1

slava77 commented:
-1 
please resolve merge conflicts before we can move on

capalmer85 commented:
I can't see the conflicts... (there weren't conflicts 6 hours ago)...

![screen shot 2017-05-16 at 2 58 16 pm](https://cloud.githubusercontent.com/assets/5417607/26123218/21958f76-3a48-11e7-9869-21b82e76ace2.png)

capalmer85 commented:
Just to be clear... I can see that there are conflicts in Calibration/TkAlCaRecoProducers/python/ALCARECOLumiPixelsMinBias_cff.py
but I can't see what they are.  We moved this file into Calibration/LumiAlCaRecoProducers but otherwise didn't edit it.

capalmer85 commented:
ok, I see the diff with some digging.  @slava77, @samhiggie is going to close the backport PR.  Should that resolve the conflict (your comment on the other one seemed to suggest) or should we import these changes by hand?
![screen shot 2017-05-16 at 3 05 12 pm](https://cloud.githubusercontent.com/assets/5417607/26123667/b5032862-3a49-11e7-89bb-c2ec8aeae043.png)

slava77 commented:
On 5/16/17 12:10 PM, capalmer85 wrote:
> ok, I see the diff with some digging. @slava77
> <https://github.com/slava77>, @samhiggie <https://github.com/samhiggie>
> is going to close the backport PR. Should that resolve the conflict
> (your comment on the other one seemed to suggest) or should we import
> these changes by hand?

This will not remove the conflict from this 92X PR

The 91X PR doesn't have a conflict because this file is unchanged there.
This means that both PRs can not go in with the same topic branch
(unless you manage to factorize the context lines changes).
Since the 92X PR needs to go firs, the 91X will have to be remade.


> screen shot 2017-05-16 at 3 05 12 pm
> <https://cloud.githubusercontent.com/assets/5417607/26123667/b5032862-3a49-11e7-89bb-c2ec8aeae043.png>
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/18233#issuecomment-301885187>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbqH77KqUHmTAgll1Eha-kjcQJN1Lks5r6fSZgaJpZM4M1O6C>.
>

davidlange6 commented:
there is no point in pursuing a 91x PR at this point...

> On May 16, 2017, at 9:33 PM, Slava Krutelyov <notifications@github.com> wrote:
> 
> On 5/16/17 12:10 PM, capalmer85 wrote:
> > ok, I see the diff with some digging. @slava77
> > <https://github.com/slava77>, @samhiggie <https://github.com/samhiggie>
> > is going to close the backport PR. Should that resolve the conflict
> > (your comment on the other one seemed to suggest) or should we import
> > these changes by hand?
> 
> This will not remove the conflict from this 92X PR
> 
> The 91X PR doesn't have a conflict because this file is unchanged there.
> This means that both PRs can not go in with the same topic branch
> (unless you manage to factorize the context lines changes).
> Since the 92X PR needs to go firs, the 91X will have to be remade.
> 
> 
> > screen shot 2017-05-16 at 3 05 12 pm
> > <https://cloud.githubusercontent.com/assets/5417607/26123667/b5032862-3a49-11e7-89bb-c2ec8aeae043.png>
> > 
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/cms-sw/cmssw/pull/18233#issuecomment-301885187>, or
> > mute the thread
> > <https://github.com/notifications/unsubscribe-auth/AEdcbqH77KqUHmTAgll1Eha-kjcQJN1Lks5r6fSZgaJpZM4M1O6C>.
> > 
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

capalmer85 committed:
Merged PCC_9_1_X from repository samhiggie

capalmer85 committed:
resolve conflict

samhiggie committed:
Merge pull request #2 from capalmer85/fix-conflict-may16

Fix conflict may16

cmsbuild commented:
Pull request #18233 was updated. @perrotta, @ghellwig, @arunhep, @cerminar, @cmsbuild, @franzoni, @slava77, @mmusich, @davidlange6 can you please check and sign again.

capalmer85 commented:
@slava77 we have edited this file by hand to match the changes, but we are still changing the location.  I thought this would resolve the conflict, but do I have to move the file back to the original location?  (can you allow this change by hand?)

slava77 commented:
the goal is to remove the conflict.

if intermediate commits/merges in this PR move and edit that file, it looks like the conflicts will persist.
interactive rebase which removes commits touching Calibration/TkAlCaRecoProducers/python/ALCARECOLumiPixelsMinBias_cff.py would help

capalmer85 committed:
Merged PCC_9_1_X from repository samhiggie

capalmer85 committed:
adding back file to try to resolve conflict

capalmer85 commented:
I can remove the conflict by adding the file back.  Is that an acceptable solution?

slava77 commented:
On 5/16/17 2:27 PM, capalmer85 wrote:
> I can remove the conflict by adding the file back. Is that an acceptable
> solution?

for me it's easier to say that you can try and see if github still
complains.
[the alternative requires manual work with your branch to check what
needs to be done]

> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/18233#issuecomment-301920801>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbhJ0kX6ZtSs7ieGDGZp6tfA83UgJks5r6hSzgaJpZM4M1O6C>.
>

samhiggie committed:
Merge pull request #3 from capalmer85/resolve-conflict-may16-v3

Resolve conflict may16 v3

cmsbuild commented:
Pull request #18233 was updated. @perrotta, @ghellwig, @arunhep, @cerminar, @cmsbuild, @franzoni, @slava77, @mmusich, @davidlange6 can you please check and sign again.

capalmer85 commented:
@slava77 that seemed to work.  

It would be great if @franzoni, @slava77, @davidlange6 could (re-)sign off very soon.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 78a49836875150879e11ea6d2c77c35a89f0fc73


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/78a49836875150879e11ea6d2c77c35a89f0fc73
https://github.com/cms-sw/cmssw/commit/6d63d8749c33af72e736deb845023948973c911e
https://github.com/cms-sw/cmssw/commit/b5e608f0c0a6a1b83a441a35e9165fef8365f2e0
https://github.com/cms-sw/cmssw/commit/53b76ecc601eb757ca2da83241b1cd5ad71eda5d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18233/19902/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18233/19902/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18233/19902/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
1000.0 step2
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step2_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

</pre>1001.0 step2
<pre>runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5/step2_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5.log

</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/78a49836875150879e11ea6d2c77c35a89f0fc73
https://github.com/cms-sw/cmssw/commit/6d63d8749c33af72e736deb845023948973c911e
https://github.com/cms-sw/cmssw/commit/b5e608f0c0a6a1b83a441a35e9165fef8365f2e0
https://github.com/cms-sw/cmssw/commit/53b76ecc601eb757ca2da83241b1cd5ad71eda5d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18233/19902/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18233/19902/git-merge-result

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

slava77 commented:
the failures are related to this PR:
```
   [0] Processing  Event run: 165121 lumi: 62 event: 23609118 stream: 0
   [1] Running path 'pathALCARECOLumiPixelsMinBias'
   [2] Calling method for module HLTHighLevel/'ALCARECOLumiPixelsMinBiasHLT'
Exception Message:
 HLTHighLevel [instance: ALCARECOLumiPixelsMinBiasHLT - path: pathALCARECOLumiPixelsMinBias]: No triggerList with key LumiPixelsMinBias in AlCaRecoTriggerBitsRcd

```

capalmer85 commented:
Yes, I see that as well.  It looks like the trigger bit was not updated
with the change of its label this in file.  (Please note... these were not
our edits.)

I'm going to see if I can made a change and get the matrix to run in my
local area with the same version of CMSSW.

On Wed, May 17, 2017 at 9:41 AM, Slava Krutelyov <notifications@github.com>
wrote:

> the failures are related to this PR:
>
>    [0] Processing  Event run: 165121 lumi: 62 event: 23609118 stream: 0
>    [1] Running path 'pathALCARECOLumiPixelsMinBias'
>    [2] Calling method for module HLTHighLevel/'ALCARECOLumiPixelsMinBiasHLT'
> Exception Message:
>  HLTHighLevel [instance: ALCARECOLumiPixelsMinBiasHLT - path: pathALCARECOLumiPixelsMinBias]: No triggerList with key LumiPixelsMinBias in AlCaRecoTriggerBitsRcd
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/18233#issuecomment-302093665>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AFKqh9l-QJVUmnw1YL67QuqSdCy9LvATks5r6vkZgaJpZM4M1O6C>
> .
>

capalmer85 commented:
I have confirmed that this is a problem with the release and not our PR.  The prescription below produces this error.  I think the error is that the trigger bit name was not changed.  I'm not sure where that is defined, but I'm going to look now.

cmsrel CMSSW_9_2_X_2017-05-16-1100
cd CMSSW_9_2_X_2017-05-16-1100/src
cmsenv
cmsDriver.py step2  --conditions auto:run1_data -s RAW2DIGI,L1Reco,RECO,EI,ALCAPRODUCER:@allForExpress,DQM:@express,ENDJOB --process RECO --data  --eventcontent ALCARECO,DQM --runUnscheduled  --scenario pp --datatier ALCARECO,DQMIO --customise Configuration/DataProcessing/RecoTLR.customiseExpress -n 100  --filein /store/data/Run2011B/MinimumBias/RAW/v1/000/177/719/BC2F8185-D4EC-E011-8370-001D09F23174.root RAW2DIGI,L1Reco,RECO,EI,ALCAPRODUCER:@allForExpress,DQM:@express,ENDJOB

capalmer85 commented:
First, I'm checking if CMSSW_9_2_X_2017-05-16-2300 has the same issue.

capalmer85 commented:
Same in 05-16-2300 version.  Key was VertexPixelZeroBias and changed to LumiPixelsMinBias.  looking...

capalmer85 commented:
Apparently the trigger bits are causing problems elsewhere too.  I'm in contact with an expert but he is currently busy with another "fire with AlCaRecoTriggerBits".  I imagine that either there will be a patch soon or that we will change the trigger bit name (in one place or another).

franzoni commented:
we don't understand yet
the origin of the problem
since the wf failing reported by Chris 
is part of the tests we did and the bot did too.

Inverstingating

franzoni commented:
understanding of what's happening, after talking to @capalmer85 :  

+ the cmsDriver @capalmer85 reports above failing prove a specific file/runnumber in the 2011 history where the hlt key being sought is not present; 
+ workflows 1000 and 1001 are covered and run to completion both in CMSSW_9_2_X_2017-05-16-1100 and CMSSW_9_2_X_2017-05-16-2300

+ the test failures of this PR stem from the fact that the modification done in the last day to Calibration/LumiAlCaRecoProducers/python/ALCARECOLumiPixelsMinBias_cff.py bring in a dependency on conditions which have entered with the PR https://github.com/cms-sw/cmssw/pull/18233 ; 
however, when this PR gets tested by doing cms-merge-topic on top of CMSSW_9_2_X_2017-05-16-1100 , the updates to autoCond.py are not applied, see [1].

At this point, a possible solution suggested to @capalmer85 is to rebase this whole set of commits on top of master.



[1]
```
cmsrel CMSSW_9_2_X_2017-05-16-1100
cd CMSSW_9_2_X_2017-05-16-1100/src
cmsenv
git cms-merge-topic 18233
git cms-addpkg Configuration/AlCa
grep data  Configuration/AlCa/python/autoCond.py
```

capalmer85 commented:
At the suggestion of @franzoni I rebased to 92X and opened a new PR:
https://github.com/cms-sw/cmssw/pull/18813

franzoni commented:
shouldn't the bot in some way
provide for itself when building the local installation to run the tests,
consistent set of sparse checkout's
which bring in all the modifications which have been merged
together with those creating conflicts ?

Dr15Jones commented:
I believe the bot does try to get all the pull requests which were merged since the last IB. However, that does constitute an unknown time at which it makes that list. I'm pretty sure that when tests are requested, one can explicitly say which other pull requests should be used.

slava77 commented:
-1 

superseded by #18813
